File: frontend/demo/component/switch/switch-basic-features.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/switch';
import '@vaadin/tooltip';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('switch-basic-features')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-switch label="Autosave" helper-text="Automatically save changes as you work">
        <vaadin-tooltip slot="tooltip" text="Last saved 5 minutes ago"></vaadin-tooltip>
      </vaadin-switch>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/switchcomponent/SwitchBasicFeatures.java
package com.vaadin.demo.component.switchcomponent;

import com.vaadin.flow.component.checkbox.Switch;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("switch-basic-features")
public class SwitchBasicFeatures extends Div {

    public SwitchBasicFeatures() {
        // tag::snippet[]
        Switch autosave = new Switch("Autosave");
        autosave.setHelperText("Automatically save changes as you work");
        autosave.setTooltipText("Last saved 5 minutes ago");

        add(autosave);
        // end::snippet[]
    }

    public static class Exporter // hidden-source-line
            extends DemoExporter<SwitchBasicFeatures> { // hidden-source-line
    } // hidden-source-line
}
